### PR TITLE
DP-1433: modify avatax project to build and deploy jar to artifactory

### PR DESCRIPTION
**Pull request by patvong**
#### Description
* fix file encoding issue and add the version in the generated jar file.
* create Jenkinsfile to build the project with Ant in Docker and publish the jar to artifactor

#### Manual merge checklist:
- [ ]  Code Review completed.

### CODE REVIEW

**Comment by patvong** [comment]
@nzin-appdirect file encoding errors during the build using ant.

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    stage('Build') {
+        echo 'Building ant project with Docker'
+        sh "docker run -it --rm -v \"\$(pwd)\":/src -e \"ANT_ARGS=-buildfile /src/avatax4j/build.xml compile jar\" brandonstevens/ant"
+
+        echo 'Displaying built jar file'
+        sh "ls -la avatax4j/build/"
+    }
+
+    stage('Artifacts') {
+        echo 'Publish artifacts'
```
Comment (adrienzieba) [inline Jenkinsfile]: not sure it should run for Pull request. should be done only for master branch

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                        [$class: 'CloneOption', noTags: false],
+                        [$class: 'LocalBranch', localBranch: "**"]
+                ]
+        ]
+        echo sh(returnStdout: true, script: 'env')
+    }
+
+    stage('Build') {
+        echo 'Building ant project with Docker'
+        sh "docker run -it --rm -v \"\$(pwd)\":/src -e \"ANT_ARGS=-buildfile /src/avatax4j/build.xml compile jar\" brandonstevens/ant"
```
Comment (adrienzieba) [inline Jenkinsfile]: I find that this docker image is too generic.
https://github.com/brandonstevens/ant-docker/blob/master/Dockerfile
Maybe we should create our own dockerfile for this build to control version of ant and java.

File: avatax4j/src/com/avalara/avatax/services/tax/GetTaxRequest.java [hunk 3]
```diff
@@ -268,17 +268,17 @@ public void setCustomerCode(java.lang.String customerCode) {
...
-        C � Tribal Government<br/>
-        D � Foreign Diplomat<br/>
-        E � Charitable Organization<br/>
-        F � Religious/Education<br/>
-        G � Resale<br/>
-        H � Agricultural Production<br/>
-        I � Industrial Prod/Mfg.<br/>
-        J � Direct Pay Permit<br/>
-        K � Direct Mail<br/>
+        A Federal Government<br/>
```
Comment (adrienzieba) [inline avatax4j/src/com/avalara/avatax/services/tax/GetTaxRequest.java]: It is weird you need to modify javadoc to make it works.

File: Jenkinsfile (hunk 1 above)
Comment (patvong) [reply Jenkinsfile]: @adrienzieba not sure what change I need to do for that?

File: Jenkinsfile (hunk 2 above)
Comment (patvong) [reply Jenkinsfile]: @adrienzieba I'm now using https://hub.docker.com/r/frekele/ant which list what java is used to build the docker ant image. Does that LGTY?

File: avatax4j/src/com/avalara/avatax/services/tax/GetTaxRequest.java (hunk 3 above)
Comment (patvong) [reply avatax4j/src/com/avalara/avatax/services/tax/GetTaxRequest.java]: Fixed by specifiying the file encoding windows-1552

File: Jenkinsfile (hunk 1 above)
Comment (adrienzieba) [reply Jenkinsfile]: something like https://github.com/AppDirect/JiraAnalyzer/blob/master/Jenkinsfile#L65

**Comment by patvong** [comment]
@adrienzieba can you review the last commit and merge if that LGTY?

**Pull request merged**